tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -377,6 +379,10 @@ def prepare_ir(argv: argparse.Namespace):
     is_tf, _, _, _, _ = deduce_legacy_frontend_by_namespace(argv)
     argv = arguments_post_parsing(argv)
     t = tm.Telemetry()
+
+    if is_tf:
+        trace_tf_model_if_needed(argv)
```

Comment:
Tracing is needed for fallback on legacy as well. 
There is a model in TF Hub https://tfhub.dev/google/yamnet/1, which works only through legacy path and in current implementation MO can fallback to legacy in some cases, which happens for this model.

But legacy pipeline needs tracing, so if we will do it only in `GraphIteator` creation method, after fallback we will have original not traced KrasLayer and legacy path will fail.

Reply:
Do we have a test to cover this case?
Also, I think we do not need fallback for conversion of models from memory.
This is quite new feature and not well formed in the previous releases due to bugs and limitations: hangs and long conversion, workaround with temporal files. Also, in this release we have plan to avoid fallback for control flow.

Reply:
Okay, as we don't need fallback for this case, I moved tracing code to `GraphIteator` creation method.